wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/AjaxLazyLoadPanel.java

```diff
@@ -139,56 +182,176 @@ protected void handleCallbackScript(final IHeaderResponse response,
+	private void setState(final LoadingState newState)
 	{
-		this.state = state;
+		this.state = newState;
 		getPage().dirty();
 	}
 
 	/**
+	 * Gets the central lazy load timer from the page if one is registered
 	 * 
-	 * @param markupId
-	 *            The components markupid.
-	 * @return The component that must be lazy created. You may call setRenderBodyOnly(true) on this
-	 *         component if you need the body only.
+	 * @return the central lazy load timer or {@code null}
 	 */
-	public abstract Component getLazyLoadComponent(String markupId);
+	private AjaxLazyLoadTimer getLazyLoadTimer()
+	{
+		return getPage().getMetaData(AJAX_LAZY_LOAD_TIMER);
+	}
 
 	/**
-	 * Called when the placeholder component is replaced with the lazy loaded one.
-	 *
-	 * @param component
-	 *      The lazy loaded component
-	 * @param target
-	 *      The Ajax request handler
+	 * Sets the central lazy load timer on the page
+	 * 
+	 * @param timer
+	 *            the timer to set
 	 */
-	protected void onComponentLoaded(Component component, AjaxRequestTarget target)
+	private void setLazyLoadTimer(AjaxLazyLoadTimer timer)
 	{
+		getPage().setMetaData(AJAX_LAZY_LOAD_TIMER, timer);
 	}
 
 	/**
-	 * @param markupId
-	 *            The components markupid.
-	 * @return The component to show while the real component is being created.
+	 * @deprecated No longer called as the AjaxLazyLoadPanel uses a central AJAX timer, and it
+	 *             doesn't make sense to modify the single timer from multiple panels
 	 */
-	public Component getLoadingComponent(final String markupId)
+	@SuppressWarnings("javadoc")
+	@Deprecated
+	protected void updateAjaxAttributes(AjaxRequestAttributes attributes)
 	{
-		IRequestHandler handler = new ResourceReferenceRequestHandler(
-			AbstractDefaultAjaxBehavior.INDICATOR);
-		return new Label(markupId, "<img alt=\"Loading...\" src=\"" +
-			RequestCycle.get().urlFor(handler) + "\"/>").setEscapeModelStrings(false);
 	}
 
+	/**
+	 * Allows subclasses to change the callback script if needed.
+	 * 
+	 * @param response
+	 *            the current response that writes to the header
+	 * @param callbackScript
+	 *            the JavaScript to write in the header
+	 * @param component
+	 *            the component which produced the callback script
+	 */
+	protected void handleCallbackScript(final IHeaderResponse response,
+		final CharSequence callbackScript, final Component component)
+	{
+		response.render(OnDomReadyHeaderItem.forScript(callbackScript));
+	}
+
+	/**
+	 * The AJAX timer for updating the AjaxLazyLoadPanel. Is designed to be a page-local singleton
+	 * keeping track of multiple LazyLoadPanels using reference counting.
+	 */
+	private static class AjaxLazyLoadTimer extends AbstractAjaxTimerBehavior
+	{
+		private static final long serialVersionUID = 1L;
+
+		private final List<AjaxLazyLoadPanel> lazyLoadPanels = new ArrayList<>();
+
+		public AjaxLazyLoadTimer()
+		{
+			super(Duration.ONE_SECOND);
```

Comment:
Or go one step futher, and make it a parent interface.
So, you just call findParent(IAjaxLazyLoadTimerSettings.class), and that way we have even more control over it, and it will still work when attached only to a page.